core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -383,6 +383,9 @@ private long catchUp(JournalReader journalReader) {
       if (entry == null) {
         return journalReader.getNextSequenceNumber();
       }
+      if (journalReader.getNextSequenceNumber() == 0) {
```

Comment:
this seems like it should have the same problem that I fixed in `UfsJournalCheckpointThread.java`. Can we check `entry.getSequenceNumber()` instead?

Reply:
addressed and cherrypicked.